packages/mui-material/src/Modal/Modal.test.js

```diff
@@ -880,4 +880,24 @@ describe('<Modal />', () => {
       );
     }).not.toErrorDev();
   });
+
+  it('should not override default onKeyDown', () => {
+    const handleKeyDown = spy();
+    const handleClose = spy();
+    const { getByTestId } = render(
+      <Modal open onKeyDown={handleKeyDown} onClose={handleClose}>
+        <div data-testid="modal" tabIndex={-1} />
+      </Modal>,
+    );
+    act(() => {
+      getByTestId('modal').focus();
+    });
+
+    fireEvent.keyDown(getByTestId('modal'), {
+      key: 'Escape',
+    });
```

Comment:
No need to focus the element, `fireEvent.keyDown` below fires the event directly on the `#modal` element.

```suggestion
    const { getByTestId } = render(
      <Modal open onKeyDown={handleKeyDown} onClose={handleClose}>
        <div data-testid="modal" tabIndex={-1} />
      </Modal>,
    );

    fireEvent.keyDown(getByTestId('modal'), {
      key: 'Escape',
    });
```

You can also use `user` instead of `fireEvent`, which seems like the option we want to use for new tests (you'll need to make the test function `async`).

```suggestion
    const { user, getByTestId } = render(
      <Modal open onKeyDown={handleKeyDown} onClose={handleClose}>
        <div data-testid="modal" tabIndex={-1} />
      </Modal>,
    );

    await user.type(getByTestId('modal'), '{Escape}');
```

Reply:
Done, updated test here https://github.com/mui/material-ui/pull/43757/commits/a475aa73fc6d33f494d83db36bc5b7211cd19faf

Note: used `keyboard` method in user object, it felt a bit simpler.

Reply:
I forgot about `keyboard`, much better.